lib/net.js

```diff
@@ -1131,6 +1203,19 @@ function lookupAndConnect(self, options) {
   }
   port |= 0;
 
+  if (typeof autoSelectFamily === 'boolean') {
+    autoSelectFamily = autoSelectFamily ? 250 : 0;
+  } else if (typeof autoSelectFamily !== 'undefined') {
+    autoSelectFamily = NumberParseInt(autoSelectFamily, 10);
+    validateInt32(autoSelectFamily, 0);
+
+    if (autoSelectFamily < 10) {
+      autoSelectFamily = 10;
+    }
+  } else {
+    autoSelectFamily = 0;
+  }
```

Comment:
nit: using a switch here avoid "recomputing" the value of `typeof autoSelectFamily`
```suggestion
  switch (typeof autoSelectFamily) {
    case 'boolean':
      autoSelectFamily = autoSelectFamily ? 250 : 0;
      break;

    case 'undefined':
      autoSelectFamily = 0;
      break;

    default:
      autoSelectFamily = NumberParseInt(autoSelectFamily, 10);
      validateInt32(autoSelectFamily, 0);

      if (autoSelectFamily < 10) {
        autoSelectFamily = 10;
      }
  }
```

Reply:
Good find! Updating!